hazelcast/src/main/java/com/hazelcast/internal/config/ConfigValidator.java

```diff
@@ -43,6 +46,11 @@
 
     private static final ILogger LOGGER = Logger.getLogger(ConfigValidator.class);
 
+    private static final EnumSet<EvictionConfig.MaxSizePolicy> SUPPORTED_ON_HEAP_NEAR_CACHE_MAXSIZE_POLICIES
+            = EnumSet.of(EvictionConfig.MaxSizePolicy.ENTRY_COUNT);
+
+    private static final boolean ENTERPRISE = BuildInfoProvider.getBuildInfo().isEnterprise();
```

Comment:
we should not cache result of `BuildInfoProvider.getBuildInfo()` in a static field. 

It leads to a hard-to-find test failures. You can either cache it in an instance field or just remove the cache completely and call `BuildInfoProvider.getBuildInfo()` whenever you need it - the `BuildInfoProvider` is caching it internally whenever it's possible. (=when overrides are disabled)